### ISSUE TITLE
install.packages doesn't throw warnings under RStudio

### ISSUE THREAD

**Issue by hughjonesd** [rstudio/rstudio#2316]
### System details

    RStudio Edition : Desktop
    RStudio Version :  1.1.423
    OS Version      :  MacOS 10.12.6
    R Version       :  3.4.3

### Steps to reproduce the problem

Run e.g.
`tryCatch(install.packages("Hmisc", repos = "BLAH"), warning = function (w) cat("Got warning"))`

### Describe the problem in detail

Under RStudio this does not catch the warning. Under R in the terminal, the warning is caught.

The problem is that install.packages calls .rs.callAs

```
get('install.packages', 'package:utils')
function (...) 
.rs.callAs(name, hook, original, ...)

```

and this function installs error handlers which print errors using `cat`, rather than rethrowing them. See https://github.com/rstudio/rstudio/blob/master/src/cpp/r/R/Tools.R

**Comment by ronblum** [comment]
@hughjonesd Just to make sure I've reproduced the problem (in my case, RStudio Mac Desktop 1.2.379-1), is this the message you get in RStudio?

> Warning in install.packages :
  unable to access index for repository BLAH/src/contrib:
  cannot open URL 'BLAH/src/contrib/PACKAGES'
Warning in install.packages :
  package ‘Hmisc’ is not available (for R version 3.4.3)
Warning in install.packages :
  unable to access index for repository BLAH/bin/macosx/el-capitan/contrib/3.4:
  cannot open URL 'BLAH/bin/macosx/el-capitan/contrib/3.4/PACKAGES'

**Comment by akgold** [comment]
As a temporary workaround, warnings from `install.packages` can be caught if you use `utils::install.packages`.
